jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/Server.java

```diff
@@ -539,12 +539,12 @@ protected void doStart() throws Exception
             if (getStopAtShutdown())
                 ShutdownThread.register(this);
 
-            //Register the Server with the handler thread for receiving
-            //remote stop commands
-            ShutdownMonitor.register(this);
-
-            //Start a thread waiting to receive "stop" commands.
-            ShutdownMonitor.getInstance().start(); // initialize
+            Shutdown shutdown = getShutdown();
```

Comment:
Just do both ShutdownMonitor.register and getBean(Shutdown.class).addComponent(this)

Reply:
Why have 2 Shutdown controlling servers active at the same time?